components/Press/PressLinks/LinkGroup.js

```diff
@@ -0,0 +1,61 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import styles from './PressLinks.css';
+import OutboundLink from '../../_common_/OutboundLink/OutboundLink';
+import Button from '../../_common_/Button/Button';
+
+class LinkGroup extends Component {
+  static propTypes = {
+    title: PropTypes.string.isRequired,
+    links: PropTypes.arrayOf(
+      PropTypes.shape({
+        url: PropTypes.string.isRequired,
+        title: PropTypes.string.isRequired,
+      }),
+    ).isRequired,
+    MaxLinks: PropTypes.number.isRequired,
+  };
+
+  state = {
+    ShowAll: false,
+  };
+
+  clickHandler = () => {
+    const { ShowAll } = this.state;
+    this.setState({ ShowAll: !ShowAll });
+  };
+
+  render() {
+    const { ShowAll } = this.state;
+    const { title, links, MaxLinks } = this.props;
+    return (
+      <div className={styles.articlesGroup}>
+        <h2>{title}</h2>
+        <ul>
+          {links.map((link, index) => {
+            if (index >= MaxLinks && !ShowAll) {
+              return null;
+            }
+            return (
+              <li key={`GroupLink_${link.url}`}>
+                <OutboundLink href={link.url}>{link.title}</OutboundLink>
+              </li>
+            );
+          })}
+        </ul>
+        {links.length > MaxLinks && (
+          <Button
+            aria-pressed={ShowAll} // this needs to be passed and accepted by the button component
```

Comment:
Okay, cool. Let's give `<Button>` the power to accept `data-` attributes and `aria-` attributes! 💪 

Stealing some code from work 🙈 - go ahead and make `common/utils/prop-utils.js` with the following content:
```javascript
// General utilities for dealing with component prop types
import pickBy from 'lodash/pickBy';

export function getPropsStartingWith(string, props) {
  return pickBy(props, (value, key) => key.startsWith(string));
}

export function getDataAttributes(props) {
  return getPropsStartingWith('data-', props);
}

export function getAriaAttributes(props) {
  return getPropsStartingWith('aria-', props);
}
```

There are also tests already written - feel free to expand upon them:
```javascript
/* eslint-env jest */
import { getPropsStartingWith, getDataAttributes, getAriaAttributes } from '../prop-utils';

describe('getPropsStartingWith', () => {
  test('should return the props that start with the given string', () => {
    const props = {
      'test-1': 'test',
      'test-2': 'test 2',
      'another-prop': 'test 3',
    };

    const result = getPropsStartingWith('test-', props);
    expect(result).toEqual({
      'test-1': 'test',
      'test-2': 'test 2',
    });
  });
});

describe('getDataAttributes', () => {
  test('should return only the props that start with `data-`', () => {
    const props = {
      'data-ci': 'ci',
      'data-gtm': 'gtm',
      children: 'test children',
    };

    const result = getDataAttributes(props);

    expect(result).toEqual({
      'data-ci': 'ci',
      'data-gtm': 'gtm',
    });
  });
});

describe('getAriaAttributes', () => {
  test('should return only the props that start with `aria-`', () => {
    const props = {
      'aria-label': 'a11y rocks!',
      'aria-checked': 'true',
      children: 'test children',
    };

    const result = getAriaAttributes(props);

    expect(result).toEqual({
      'aria-label': 'a11y rocks!',
      'aria-checked': 'true',
    });
  });
});
```

Once you have those utilities, you can expand Button like so:
```javascript
propTypes = { ...whatWasThereBefore, datum: PropTypes.any };
defaultProps = { ...whatWasThereBefore, datum: '' };

...

render() {
  const customDataAttributes = getDataAttributes(props);
  const ariaAttributes = getAriaAttributes(props);

  // and then you can spread the above variables into the attributes of `button`
}
```

Reply:
This is great (stealing this for myself).
I'm not 100% on why the propTypes and defaultProps is necessary, specifically "datum" ??

Reply:
Not much to expand on for the tests, 100% and does whats needed.

Reply:
datum is essentially a placeholder for non-reserved prop names (like children)

so, if Button gets passed a prop of `abbazabba` it will be within `datum`, but our utility functions ensure it doesn't get passed to the final html.